handlers.py

```diff
@@ -0,0 +1,115 @@
+from telegram import Location, Update
+from telegram.ext import CallbackContext, CommandHandler, MessageHandler, filters
+from constants import RESTAURANT
+from decorators import check_start, user_started
+from wolt_services import check_if_restaurant_is_open
+
+restaurant_index: int | None = None
+restaurant_is_open: bool | str = False
+lat: float | None = None
+lon: float | None = None
+
+restaurant_list = "\n".join(
+    [f"{i+1}. {name}" for i, name in enumerate(RESTAURANT.keys())]
+)
+
+
+async def handle_start(update: Update, context: CallbackContext) -> None:
+    user = update.message.from_user
+    user_started[user.id] = True
+    message = f"Hi {user.first_name}, Which restaurant should we check for you?\n{restaurant_list}"
+    await context.bot.send_message(update.effective_chat.id, message)
+
+
+@check_start
+async def handle_location(update: Update, context: CallbackContext) -> None:
+    global restaurant_is_open, lat, lon
+    location: Location | None = (
+        update.message.location
+        if update.message
+        else update.edited_message.location
+        if update.edited_message
+        else None
+    )
+
+    if location is None:
+        return
+
+    lat = location.latitude
+    lon = location.longitude
+
+    if restaurant_index is None:
+        await context.bot.send_message(
+            update.effective_chat.id,
+            f"Received your location: {lat}, {lon}, now choose your restaurant\n{restaurant_list}",
+        )
+        return
+
+    restaurant_name = list(RESTAURANT.keys())[restaurant_index]
+    restaurant_id = RESTAURANT[restaurant_name]
+    if restaurant_is_open:
+        restaurant_is_open = check_if_restaurant_is_open(restaurant_id, lat, lon)
+        return
+
+    restaurant_is_open = check_if_restaurant_is_open(restaurant_id, lat, lon)
+
+    if type(restaurant_is_open) is str:
+        await context.bot.send_message(
+            update.effective_chat.id,
+            "oops, seems like we had some issues getting the data, try to send your location again",
+        )
+
+    await context.bot.send_message(
+        update.effective_chat.id,
+        f"{restaurant_name} is {"open" if restaurant_is_open else "close"}",
+    )
+
+
+@check_start
+async def handle_choose_restaurant(update: Update, context: CallbackContext) -> None:
```

Comment:
Function name is not clear enough (give your functions names the indcate their purpose + without spelling mistakes).